deepchecks/vision/vision_data/vision_data.py

```diff
@@ -0,0 +1,262 @@
+# Copyright (C) 2021-2022 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Module containing the VisionData class and its functions."""
+import typing as t
+from collections import defaultdict
+
+from deepchecks.core.errors import DeepchecksValueError, ValidationError
+from deepchecks.vision.vision_data import TaskType
+from deepchecks.vision.vision_data.format_validators import (validate_additional_data_format,
+                                                             validate_embeddings_format,
+                                                             validate_image_identifiers_format, validate_images_format,
+                                                             validate_labels_format, validate_predictions_format)
+from deepchecks.vision.vision_data.utils import (BatchOutputFormat, get_class_ids_from_numpy_labels,
+                                                 get_class_ids_from_numpy_preds, shuffle_loader)
+
+VD = t.TypeVar('VD', bound='VisionData')
+
+
+class VisionData:
+    """VisionData is the base data object of deepcheck vision used as input to checks and suites.
+
+    Parameters
+    ----------
+    batch_loader :
+        A batch_loader which load a batch of data in an iterative manner. Batch loader batch output must be a
+        dictionary in BatchOutputFormat format. The batch loader must provide SHUFFLED batches.
+    task_type : str
+        The task type of the data. can be one of the following: 'classification', 'semantic_segmentation',
+        'object_detection', 'other'. For 'other', only image related checks (such as ImagePropertyOutliers) will be run.
+    label_map : Dict[int, str], optional
+        A dictionary mapping class ids to their names.
+    dataset_name: str, optional
+        Name of the dataset to use in the displays instead of "Train" or "Test".
+    shuffle_batch_loader: bool, default=True
+        If True and the batch loader is of known type that can be shuffled, it will be shuffled.
+    """
+
+    def __init__(
+            self,
+            batch_loader,
+            task_type: str,
```

Comment:
let the people use the enum too, you can also define the enum as:
TaskType(str, enum) 
so the only change you need here I think is in the typings of the signature but check me on that